Add examples to RRef doc

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30516 Add examples to RRef doc**
* #30515 Make doc source format consistent in rpc/init.cpp
* #30491 Reorganize rpc API doc and add introduction

Differential Revision: [D18728183](https://our.internmc.facebook.com/intern/diff/D18728183)

mrshenli committed:
Add examples to RRef doc

[ghstack-poisoned]

rohan-varma commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -72,6 +72,34 @@ PyObject* rpc_init(PyObject* /* unused */) {
+
+              >>> import torch.distributed.rpc as rpc
+              >>> rref = rpc.remote("worker1", torch.add, args=(torch.ones(2), 3))
+
```
nit: maybe add a ``rref.to_here()`` to show how we can get the value?

rohan-varma reviewed (approved):
LGTM

mrshenli committed:
Update on "Add examples to RRef doc"


Differential Revision: [D18728183](https://our.internmc.facebook.com/intern/diff/D18728183)

[ghstack-poisoned]

mrshenli commented:
Test failures are irrelevant. Landing

```
16:56:43 ======================================================================
16:56:43 FAIL: test_numerical_consistency_per_tensor (__main__.TestFakeQuantizePerTensor)
16:56:43 Comparing numerical consistency between CPU quantize/dequantize op and the CPU fake quantize op
16:56:43 ----------------------------------------------------------------------
16:56:43 Traceback (most recent call last):
16:56:43   File "test_fake_quant.py", line 112, in test_numerical_consistency_per_tensor
16:56:43     @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
16:56:43   File "C:\Jenkins\Miniconda3\lib\site-packages\hypothesis\core.py", line 1067, in wrapped_test
16:56:43     raise the_error_hypothesis_found
16:56:43   File "test_fake_quant.py", line 128, in test_numerical_consistency_per_tensor
16:56:43     np.testing.assert_allclose(Y, Y_prime.cpu(), rtol=tolerance, atol=tolerance)
16:56:43   File "C:\Jenkins\Miniconda3\lib\site-packages\numpy\testing\_private\utils.py", line 1515, in assert_allclose
16:56:43     verbose=verbose, header=header, equal_nan=equal_nan)
16:56:43   File "C:\Jenkins\Miniconda3\lib\site-packages\numpy\testing\_private\utils.py", line 841, in assert_array_compare
16:56:43     raise AssertionError(msg)
16:56:43 AssertionError: 
16:56:43 Not equal to tolerance rtol=1e-06, atol=1e-06
```

```
Nov 28 17:13:29 ======================================================================
Nov 28 17:13:29 FAIL [1.124s]: test_backward_per_tensor (__main__.TestFakeQuantizePerTensor)
Nov 28 17:13:29 Tests the backward method.
Nov 28 17:13:29 ----------------------------------------------------------------------
Nov 28 17:13:29 Traceback (most recent call last):
Nov 28 17:13:29   File "test_fake_quant.py", line 89, in test_backward_per_tensor
Nov 28 17:13:29     @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
Nov 28 17:13:29   File "/var/lib/jenkins/.local/lib/python3.6/site-packages/hypothesis/core.py", line 1067, in wrapped_test
Nov 28 17:13:29     raise the_error_hypothesis_found
Nov 28 17:13:29   File "test_fake_quant.py", line 109, in test_backward_per_tensor
Nov 28 17:13:29     np.testing.assert_allclose(dX.cpu(), X.grad.cpu().detach().numpy(), rtol=tolerance, atol=tolerance)
Nov 28 17:13:29   File "/opt/conda/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1515, in assert_allclose
Nov 28 17:13:29     verbose=verbose, header=header, equal_nan=equal_nan)
Nov 28 17:13:29   File "/opt/conda/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 841, in assert_array_compare
Nov 28 17:13:29     raise AssertionError(msg)
Nov 28 17:13:29 AssertionError: 
Nov 28 17:13:29 Not equal to tolerance rtol=1e-06, atol=1e-06
```
